src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -1,46 +1,63 @@
 package roomescape.controller;
 
-import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.*;
-import roomescape.dto.ReservationDto;
-import roomescape.domain.Reservation;
-
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import roomescape.domain.Reservation;
+import roomescape.dto.ReservationCreateRequest;
 
-@Controller
+@RestController
 @RequestMapping("reservations")
 public class ReservationController {
 
     private final List<Reservation> reservations = new ArrayList<>();
     private final AtomicLong index = new AtomicLong(1);
 
-    @GetMapping("")
+    @GetMapping
     public ResponseEntity<List<Reservation>> reservations() {
         return ResponseEntity.ok()
-                             .body(reservations);
+                .body(reservations);
     }
 
-    @PostMapping("")
-    public ResponseEntity<Reservation> createReservation(@RequestBody final ReservationDto reservationDto) {
-        Reservation reservation = Reservation.toEntity(index.getAndIncrement(), reservationDto);
+    @PostMapping
+    public ResponseEntity<Reservation> createReservation(@RequestBody ReservationCreateRequest reservationCreateRequest) {
+        Reservation reservation = Reservation.toEntity(
+                index.getAndIncrement(),
+                reservationCreateRequest.name(),
+                reservationCreateRequest.date(),
+                reservationCreateRequest.time());
```

Comment:
```suggestion
        Reservation reservation = new Reservation(
                reservationCreateRequest.name(),
                reservationCreateRequest.date(),
                reservationCreateRequest.time());
```
정적 팩터리 메서드가 정말 유의미하게 사용되고 있는지 잘 모르겠습니다. 만들어두신 생성자만 사용해도 되지 않을까요?

Reply:
음 굳이 정적 팩토리 메서드를 사용하지 않아도 되겠네요. 이 메서드는 일단 삭제하겠습니다!